tensorflow/c/experimental/filesystem/plugins/gcs/gcs_filesystem_test.cc

```diff
@@ -12,50 +12,147 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
-#include "tensorflow/c/experimental/filesystem/filesystem_interface.h"
+#include "tensorflow/c/experimental/filesystem/plugins/gcs/gcs_filesystem.h"
+
+#include <random>
+
 #include "tensorflow/c/tf_status_helper.h"
+#include "tensorflow/core/platform/path.h"
 #include "tensorflow/core/platform/stacktrace_handler.h"
 #include "tensorflow/core/platform/test.h"
 
-#define ASSERT_TF_OK(x) ASSERT_EQ(TF_OK, TF_GetCode(x))
+#define ASSERT_TF_OK(x) ASSERT_EQ(TF_OK, TF_GetCode(x)) << TF_Message(x)
 
-// Forward declaration
-namespace tf_gcs_filesystem {
-void Init(TF_Filesystem* filesystem, TF_Status* status);
-}
+static const char* content = "abcdefghijklmnopqrstuvwxyz1234567890";  // 36
+
+namespace gcs = google::cloud::storage;
 
 namespace tensorflow {
 namespace {
 
 class GCSFilesystemTest : public ::testing::Test {
  public:
   void SetUp() override {
+    root_dir_ = io::JoinPath(
+        tmp_dir_,
+        ::testing::UnitTest::GetInstance()->current_test_info()->name());
     status_ = TF_NewStatus();
     filesystem_ = new TF_Filesystem;
     tf_gcs_filesystem::Init(filesystem_, status_);
-    ASSERT_TF_OK(status_) << "Can not initialize filesystem. "
+    ASSERT_TF_OK(status_) << "Could not initialize filesystem. "
                           << TF_Message(status_);
   }
   void TearDown() override {
     TF_DeleteStatus(status_);
-    // TODO(vnvo2409): Add filesystem cleanup
+    tf_gcs_filesystem::Cleanup(filesystem_);
     delete filesystem_;
   }
 
+  static bool InitializeTmpDir() {
+    // This env should be something like `gs://bucket/path`
+    const char* test_dir = getenv("GCS_TEST_TMPDIR");
+    if (test_dir != nullptr) {
+      char* bucket;
+      char* object;
+      TF_Status* status = TF_NewStatus();
+      ParseGCSPath(test_dir, true, &bucket, &object, status);
+      if (TF_GetCode(status) != TF_OK) {
+        free(bucket);
+        free(object);
+        TF_DeleteStatus(status);
+        return false;
+      }
+      free(bucket);
+      free(object);
+      TF_DeleteStatus(status);
```

Comment:
```suggestion
      char* bucket;
      char* object;
      TF_Status* status = TF_NewStatus();
      ParseGCSPath(test_dir, true, &bucket, &object, status);
      free(bucket);
      free(object);
      if (TF_GetCode(status) != TF_OK) {
        TF_DeleteStatus(status);
        return false;
      }
      TF_DeleteStatus(status);
```

Reply:
This will be changed when I change `ParseGCSPath`